api/operator/v1alpha1/vmdistributedcluster_types.go

```diff
@@ -0,0 +1,372 @@
+	// +optional
+	Paused bool `json:"paused,omitempty"`
+}
+
+// +k8s:openapi-gen=true
+// ZoneSpec is a list of VMCluster instances to update.
+type ZoneSpec struct {
+	// Each VMClusterRefOrSpec is either defining a new inline VMCluster or referencing an existing one.
+	VMClusters []VMClusterRefOrSpec `json:"vmclusters,omitempty"`
+}
+
+// +k8s:openapi-gen=true
+// VMClusterRefOrSpec is either a reference to existing VMCluster or a specification of a new VMCluster.
+// +kubebuilder:validation:Xor=Ref,Spec
+type VMClusterRefOrSpec struct {
+	// Ref points to the VMCluster object.
+	// If Ref is specified, Name and Spec are ignored.
+	// +optional
+	Ref *corev1.LocalObjectReference `json:"ref,omitempty"`
+	// OverrideSpec specifies an override to the VMClusterSpec of the referenced object.
+	// This override is applied to the referenced object if `Ref` is specified.
+	// This field is ignored if `Spec` is specified.
+	// +kubebuilder:validation:Type=object
+	// +kubebuilder:validation:XPreserveUnknownFields
+	// +optional
+	OverrideSpec *apiextensionsv1.JSON `json:"overrideSpec,omitempty"`
+
+	// Name specifies the static name to be used for the VMCluster when Spec is provided.
+	// This field is ignored if `Ref` is specified.
+	// +optional
+	Name string `json:"name,omitempty"`
+	// Spec defines the desired state of a new VMCluster.
+	// This field is ignored if `Ref` is specified.
+	// +optional
+	Spec *vmv1beta1.VMClusterSpec `json:"spec,omitempty"`
+}
+
+// +k8s:openapi-gen=true
+// VMAgentNameAndSpec is a name and a specification of a new VMAgent.
+type VMAgentNameAndSpec struct {
+	// Name specifies the static name to be used for the VMAgent when Spec is provided.
+	// +optional
+	Name string `json:"name,omitempty"`
+	// Spec defines the desired state of a new VMAgent.
+	// Note that RemoteWrite and RemoteWriteSettings are ignored as its managed by the operator.
+	// +optional
+	Spec *CustomVMAgentSpec `json:"spec,omitempty"`
+	// Optional tenant ID. Only required with the OAuth authentication method.
+	// +optional
+	TenantID *string `json:"tenantID,omitempty"`
+}
+
+// +k8s:openapi-gen=true
+// CustomVMAgentSpec is a customized specification of a new VMAgent.
+// It includes selected options from the original VMAgentSpec.
+type CustomVMAgentSpec struct {
+	vmv1beta1.CommonVMAgentSpec `json:",inline"`
+	// RemoteWrite list of victoria metrics /some other remote write system
+	// for vm it must looks like: http://victoria-metrics-single:8428/api/v1/write
+	// or for cluster different url
+	// https://docs.victoriametrics.com/victoriametrics/vmagent/#splitting-data-streams-among-multiple-systems
+	// +optional
+	RemoteWrite []CustomVMAgentRemoteWriteSpec `json:"remoteWrite"`
+}
+
+// CustomVMAgentRemoteWriteSpec is a copy of VMAgentRemoteWriteSpec, but allows empty URLs
+// These urls will be overwritten by the controller
+// +k8s:openapi-gen=true
+type CustomVMAgentRemoteWriteSpec struct {
```

Comment:
Let's remove `CustomVMAgentRemoteWriteSpec` and do not allow it for now. I don't really know practical use-cases for it.

 If it's needed, remoteWrite.url could be tweaked via `spec.vmagent.extraArgs`.